sql/moz-fx-data-shared-prod/udf_js/snake_case_columns/udf.sql

```diff
@@ -0,0 +1,69 @@
+CREATE OR REPLACE FUNCTION udf_js.snake_case_columns(input ARRAY<STRING>)
+RETURNS ARRAY<STRING> DETERMINISTIC
+LANGUAGE js
+AS
+  """
+const REV_WORD_BOUND_PAT = new RegExp(
+  "\\b" + // standard word boundary
+  "|(?<=[a-z][A-Z])(?=\\d*[A-Z])" + // A7Aa -> A7|Aa boundary
+  "|(?<=[a-z][A-Z])(?=\\d*[a-z])" + // a7Aa -> a7|Aa boundary
+  "|(?<=[A-Z])(?=\\d*[a-z])"     // a7A -> a7|A boundary
```

Comment:
IMO prefixing the `+` operators would be more readable.
```suggestion
  "\\b"  // standard word boundary
  + "|(?<=[a-z][A-Z])(?=\\d*[A-Z])"  // A7Aa -> A7|Aa boundary
  + "|(?<=[a-z][A-Z])(?=\\d*[a-z])"  // a7Aa -> a7|Aa boundary
  + "|(?<=[A-Z])(?=\\d*[a-z])"  // a7A -> a7|A boundary
```